packages/opentelemetry-plugin-grpc/src/grpc.ts

```diff
@@ -122,24 +127,28 @@ export class GrpcPlugin extends BasePlugin<grpc> {
     }
   }
 
-  private _getSpanContext(metadata: grpcTypes.Metadata): SpanContext | null {
-    const metadataValue = metadata.getMap()[GRPC_TRACE_KEY] as Buffer;
-    // Entry doesn't exist
-    if (!metadataValue) {
-      return null;
-    }
-    return this._tracer.getBinaryFormat().fromBytes(metadataValue);
+  private _getSpanContext(
+    metadata: grpcTypes.Metadata
+  ): SpanContext | undefined {
+    return getExtractedSpanContext(
+      this._tracer.getHttpTextFormat().extract(Context.TODO, metadata.getMap())
+    );
   }
 
   private _setSpanContext(
     metadata: grpcTypes.Metadata,
     spanContext: SpanContext
   ): void {
-    const serializedSpanContext = this._tracer
-      .getBinaryFormat()
-      .toBytes(spanContext);
-    const buffer = Buffer.from(serializedSpanContext);
-    metadata.set(GRPC_TRACE_KEY, buffer);
+    const carrier = {};
+    this._tracer
+      .getHttpTextFormat()
+      .inject(
+        setExtractedSpanContext(Context.ROOT_CONTEXT, spanContext),
```

Comment:
In the future we will not be manipulating Context directly like this, so I added the TODO placeholder so we can easily find places in the codebase where this was done.

```suggestion
        setExtractedSpanContext(Context.TODO, spanContext),
```